source/solvers/navier_stokes_base.cc

```diff
@@ -899,12 +901,27 @@ NavierStokesBase<dim, VectorType, DofsType>::refine_mesh_kelly()
   std::vector<bool> global_refine_flags(dim * tria.n_active_cells(), false);
   std::vector<bool> global_coarsen_flags(dim * tria.n_active_cells(), false);
 
-  bool first_variable(true);
+  bool   first_variable(true);
+  double coarsening_factor = mesh_controller.calculate_coarsening_factor(
```

Comment:
```suggestion
  const double coarsening_factor = mesh_controller.calculate_coarsening_factor(
```

if it's not modified afterwards